lib/postgrex/extensions/polygon.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Postgrex.Extensions.Polygon do
+  @moduledoc false
+  import Postgrex.BinaryUtils, warn: false
+  use Postgrex.BinaryExtension, send: "poly_send"
+  alias Postgrex.Extensions.Polygon
+  alias Postgrex.Extensions.Point
+
+  def encode(_) do
+    quote location: :keep do
+      %Postgrex.Polygon{vertices: vertices} when is_list(vertices) ->
+        len = <<length(vertices)::int32>>
+        vert = Enum.map(vertices, &Point.encode_point(&1, Postgrex.Polygon))
+
+        # 32 bits for len, 64 for each x and each y
+        nbytes = 4 + 16 * length(vertices)
+
+        [<<nbytes :: int32>>, len, vert]
```

Comment:
Saves couple of words memory to put trailing binary as improper list: `len | vert]`. Please do elsewhere.